sdk/management/docs/AUTH.md

```diff
@@ -0,0 +1,92 @@
+# Authentication in Azure Management Library for Java
+
+To use the APIs in the Azure Management Library for Java, as the first step you need to create an authenticated client. This document is to introduce several possible approaches for authentication.
+
+## Prerequisites
+
+Before authenticating the Azure client, there are a few authentication classes you need to know before you decide which kind of authentication would meet your requirement.
+
+* TokenCredential
+  * The `TokenCredential` is an interface in the `azure-core` package for credentials that can provide a token. 
+  * Azure Identity offers multiple implementations of the `TokenCredential` class in the `azure-identity` package. To learn more, see [credentials in Azure Identity](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity#credentials).
+
+Sample code to create a simple `ClientSecretCredential`:
```

Comment:
I think most common one will be `UserPasswordCredential` or `InteractiveBrowserCredential`. Otherwise, we need to add a guide for how to create a service principal.

Reply:
In track 1, there isn't any introduction for the [credentials from client runtime](https://github.com/Azure/autorest-clientruntime-for-java/tree/master/azure-client-authentication) then we have AUTH.md introducing credentials. However, in track 2, the Azure Identity offers detailed description about [credentials](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/identity/azure-identity#credentials). 
So I'm thinking about to focus on introducing the authentication flow in AUTH.md when using our track 2 library, while the link to Azure Identity would help user get details about TokenCrdential implementations.

As auth file is recommended in track 1, the equivalent credentials are `ClientSecretCredential` and `ClientCertificateCredential`. That's why I put sample code for `ClientSecretCredential`.

Reply:
Then, I think it would better to have a link to service principal registration.

Reply:
Agree. I think we should point to somewhere for guide to create service principal that is used for `ClientSecretCredentialBuilder`. I could not easily find it from this md or azure-identity md (maybe I only missed it).

Reply:
https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal